app/reducers/misc.js

```diff
@@ -102,208 +131,302 @@ export function blipUrls(state = {}, action) {
   switch (action.type) {
-    case actionTypes.LOGIN_FAILURE:
-    case actionTypes.LOGIN_SUCCESS:
+    case types.LOGIN_FAILURE:
+    case types.LOGIN_SUCCESS:
       return false;
-    case actionTypes.LOGIN_REQUEST:
+    case types.LOGIN_REQUEST:
       return true;
     default:
       return state;
   }
 }
 
-function initializingApp(state = true, action) {
+function initializingApp(state = initialState.initializingApp, action) {
   switch (action.type) {
-    case actionTypes.INIT_APP_FAILURE:
-    case actionTypes.INIT_APP_SUCCESS:
+    case types.INIT_APP_FAILURE:
+    case types.INIT_APP_SUCCESS:
       return false;
-    case actionTypes.INIT_APP_REQUEST:
+    case types.INIT_APP_REQUEST:
       return true;
     default:
       return state;
   }
 }
 
-function uploading(state = false, action) {
+function uploading(state = initialState.uploading, action) {
   switch (action.type) {
-    case actionTypes.UPLOAD_REQUEST:
+    case types.UPLOAD_REQUEST:
       return true;
-    case actionTypes.READ_FILE_ABORTED:
-    case actionTypes.READ_FILE_FAILURE:
-    case actionTypes.UPLOAD_FAILURE:
-    case actionTypes.UPLOAD_SUCCESS:
-    case actionTypes.UPLOAD_CANCELLED:
-      return false;
+    case types.READ_FILE_ABORTED:
+    case types.READ_FILE_FAILURE:
+    case types.UPLOAD_FAILURE:
+    case types.UPLOAD_SUCCESS:
+    case types.UPLOAD_CANCELLED:
+      return initialState.uploading;
     default:
       return state;
   }
 }
 
-function checkingElectronUpdate(state = false, action) {
+function checkingElectronUpdate(state = initialState.checkingElectronUpdate, action) {
   switch (action.type) {
-    case actionTypes.CHECKING_FOR_UPDATES:
-    case actionTypes.AUTO_UPDATE_CHECKING_FOR_UPDATES:
-    case actionTypes.MANUAL_UPDATE_CHECKING_FOR_UPDATES:
+    case types.CHECKING_FOR_UPDATES:
+    case types.AUTO_UPDATE_CHECKING_FOR_UPDATES:
+    case types.MANUAL_UPDATE_CHECKING_FOR_UPDATES:
       return true;
-    case actionTypes.UPDATE_AVAILABLE:
-    case actionTypes.UPDATE_NOT_AVAILABLE:
-    case actionTypes.AUTOUPDATE_ERROR:
-      return false;
+    case types.UPDATE_AVAILABLE:
+    case types.UPDATE_NOT_AVAILABLE:
+    case types.AUTOUPDATE_ERROR:
+      return initialState.checkingElectronUpdate;
     default:
       return state;
   }
 }
 
-function checkingDriverUpdate(state = false, action) {
+function checkingDriverUpdate(state = initialState.checkingDriverUpdate, action) {
   switch (action.type) {
-    case actionTypes.CHECKING_FOR_DRIVER_UPDATE:
+    case types.CHECKING_FOR_DRIVER_UPDATE:
       return true;
-    case actionTypes.DRIVER_UPDATE_AVAILABLE:
-    case actionTypes.DRIVER_UPDATE_NOT_AVAILABLE:
-      return false;
+    case types.DRIVER_UPDATE_AVAILABLE:
+    case types.DRIVER_UPDATE_NOT_AVAILABLE:
+      return initialState.checkingDriverUpdate;
     default:
       return state;
   }
 }
 
+//TODO: these should be updated to blip's `working` format for consistency
```

Comment:
It wasn't, really. This is more of a note that I noticed the difference in how blip handles this vs. how uploader handles it, but since I wasn't modifying any of these reducers specifically, I didn't feel like it was in the scope of the current task to try and migrate them.